src/bootstrap/test.rs

```diff
@@ -704,6 +704,7 @@ impl Step for Miri {
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub struct CompiletestTest {
+    stage: u32,
```

Comment:
i don't think this change is necessary — i'd prefer to keep using builder.top_stage in run().